hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,395 @@
+   */
+  private boolean checkCurrentMetadataLocation(String newMetadataLocation) {
+    ViewMetadata metadata = refresh();
+    return newMetadataLocation.equals(metadata.metadataFileLocation());
+  }
+
+  private void setHmsTableParameters(
+      String newMetadataLocation, Table tbl, ViewMetadata metadata, Set<String> obsoleteProps) {
+    Map<String, String> parameters =
+        Optional.ofNullable(tbl.getParameters()).orElseGet(Maps::newHashMap);
+
+    // push all Iceberg view properties into HMS
+    metadata.properties().entrySet().stream()
+        .filter(entry -> !entry.getKey().equalsIgnoreCase(HiveCatalog.HMS_TABLE_OWNER))
+        .forEach(entry -> parameters.put(entry.getKey(), entry.getValue()));
+    if (metadata.uuid() != null) {
+      parameters.put("uuid", metadata.uuid());
+    }
+
+    // remove any props from HMS that are no longer present in Iceberg view props
+    obsoleteProps.forEach(parameters::remove);
+
+    parameters.put(
+        BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        ICEBERG_VIEW_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(BaseMetastoreTableOperations.METADATA_LOCATION_PROP, newMetadataLocation);
+
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(
+          BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+
+    // If needed set the 'storage_handler' property to enable query from Hive
+    parameters.remove(hive_metastoreConstants.META_TABLE_STORAGE);
+
+    setSchema(metadata.schema(), parameters);
+    tbl.setParameters(parameters);
+  }
+
+  private static boolean hiveLockEnabled(Configuration conf) {
+    return conf.getBoolean(ConfigProperties.LOCK_HIVE_ENABLED, true);
+  }
+
+  private Table newHMSView(ViewMetadata metadata) {
+    final long currentTimeMillis = System.currentTimeMillis();
+    String hmsTableOwner =
+        PropertyUtil.propertyAsString(
+            metadata.properties(), HiveCatalog.HMS_TABLE_OWNER, HiveHadoopUtil.currentUser());
+    Preconditions.checkNotNull(hmsTableOwner, "'hmsOwner' parameter can't be null");
+    String sqlQuery = sqlFor(metadata);
+
+    return new Table(
+        table(),
+        database(),
+        hmsTableOwner,
+        (int) currentTimeMillis / 1000,
+        (int) currentTimeMillis / 1000,
+        Integer.MAX_VALUE,
+        null,
+        Collections.emptyList(),
+        Maps.newHashMap(),
+        sqlQuery,
+        sqlQuery,
+        tableType().name());
+  }
+
+  private String sqlFor(ViewMetadata metadata) {
+    SQLViewRepresentation closest = null;
+    for (ViewRepresentation representation : metadata.currentVersion().representations()) {
+      if (representation instanceof SQLViewRepresentation) {
+        SQLViewRepresentation sqlViewRepresentation = (SQLViewRepresentation) representation;
+        if (sqlViewRepresentation.dialect().equalsIgnoreCase("hive")) {
```

Comment:
Maybe I'm missing it, but shouldn't there be code that sets `hive` as the dialect somewhere?

Reply:
I think it's engine responsibility to set the dialect. something [like](https://github.com/apache/iceberg/blob/f460964e78dbbfbe81fd093d436dca80c72e7124/core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java#L1422) 
``` 
 T withQuery(String dialect, String sql);
 ``` 

WDYT ?